packages/snaps-cli/src/webpack/config.ts

```diff
@@ -265,10 +266,11 @@ export async function getDefaultConfiguration(
 
       /**
        * The `EnvironmentPlugin` is a Webpack plugin that adds environment
-       * variables to the bundle. We use it to add the `NODE_ENV` and `DEBUG`
-       * environment variables.
+       * variables to the bundle. We use it to add the `NODE_DEBUG`, `NODE_ENV`,
+       * and `DEBUG` environment variables, as well as any custom environment
+       * variables.
        */
-      new EnvironmentPlugin(config.environment),
+      new DefinePlugin(getEnvironmentVariables(config.environment)),
```

Comment:
`EnvironmentPlugin` was causing compiler warnings. Switching to `DefinePlugin` seems to have resolved this issue.